dvc/remote/gdrive.py

```diff
@@ -41,10 +41,43 @@ def __init__(self, path):
 def gdrive_retry(func):
     retry_re = re.compile(r"HttpError (403|500|502|503|504)")
 
+    def retry_error(exc):
+        import json
+
+        if RemoteGDrive.corpora != "drive" or not exc.args:
+            return True
+        http_error = exc.args[0]
+        content = json.loads(http_error.content)
+        if content and "error" not in content:
+            return True
+        err = content["error"]
+        if (
+            err["code"] == 403
+            and len(err["errors"])
+            and err["errors"][0].get("location", "") == "file.permissions"
+        ):
+            raise DvcException(
+                "Insufficient permissions to {}. You should have {} "
+                "access level for the used shared drive. More details "
+                "at {}".format(
+                    "move the file into Trash"
+                    if RemoteGDrive.trash_only
+                    else "permanently delete the file",
+                    "Manager or Content Manager"
+                    if RemoteGDrive.trash_only
+                    else "Manager",
+                    "https://support.google.com/a/answer/7337554?hl=en",
+                )
+            ) from exc
+        return True
+
     def should_retry(exc):
         from pydrive2.files import ApiRequestError
 
-        return isinstance(exc, ApiRequestError) and retry_re.search(str(exc))
+        if not isinstance(exc, ApiRequestError):
+            return False
+
+        return retry_error(exc) and retry_re.search(str(exc))
```

Comment:
can we completely get rid of `retry_re.search(str(exc))`?

Reply:
Done, it checks numeric code now